var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -23,27 +24,28 @@ class PyNumba(PythonPackage):
     version("0.48.0", sha256="9d21bc77e67006b5723052840c88cc59248e079a907cc68f1a1a264e1eaba017")
     version("0.40.1", sha256="52d046c13bcf0de79dbfb936874b7228f141b9b8e3447cc35855e9ad3e12aa33")
 
-    depends_on("python@3.7:3.10", when="@0.55.0:", type=("build", "run"))
+    depends_on("python@3.8:3.12", when="@0.57", type=("build", "run"))
+    depends_on("python@3.7:3.11", when="@0.56", type=("build", "run"))
+    depends_on("python@3.7:3.10", when="@0.55", type=("build", "run"))
     depends_on("python@3.7:3.9", when="@0.54", type=("build", "run"))
     depends_on("python@3.6:3.9", when="@0.53", type=("build", "run"))
     depends_on("python@3.6:3.8", when="@0.52", type=("build", "run"))
-    # set upper bound for python the same as newer release
     depends_on("python@3.6:3.8", when="@0.48:0.51", type=("build", "run"))
     depends_on("python@3.3:3.7", when="@0.40.1:0.47", type=("build", "run"))
-    depends_on("py-numpy@1.18:1.23", when="@0.56.1:", type=("build", "run"))
+    depends_on("py-numpy@1.21:1.25", when="@0.57.0:", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-numpy@1.21:1.24", when="@0.57:", type=("build", "run"))
```
The upper bounds are exclusive here as well (see [setup.py#L371](https://github.com/numba/numba/blob/0.57.0/setup.py#L371))